Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM ruby:2.2.2
+
+RUN apt-get update
+RUN apt-get install -y wget apt-transport-https
+RUN wget -qO- https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
+RUN echo 'deb https://deb.nodesource.com/node_0.12 trusty main' > /etc/apt/sources.list.d/nodesource.list
+RUN apt-get update && apt-get install -y nodejs
+
+RUN mkdir /app
+WORKDIR /app
+
+# Mostly static
+ADD config.ru /app/
+ADD Rakefile /app/
+ADD bin /app/bin
+ADD public /app/public
+ADD db /app/db
+ADD .env.bootstrap /app/.env
+
+# Gems
+ADD Gemfile /app/
+ADD Gemfile.lock /app/
+ADD vendor/cache /app/vendor/cache
+ADD package.json /app/package.json
+
+# Plugins need to be added before bundling
+# because they're loaded as gems
+ADD plugins /app/plugins
+
+RUN npm install
+
+RUN bundle install --without test sqlite postgres --quiet --local --jobs 4 || bundle check
+
+# Code
+ADD config /app/config
+ADD app /app/app
+ADD lib /app/lib
+
+RUN DATABASE_URL=mysql2://user:pass@127.0.0.1/null RAILS_ENV=development PRECOMPILE=1 bundle exec rake --trace assets:precompile
+
+EXPOSE 3000
+
+CMD DATABASE_URL=$MYSQL_URL bundle exec rails server --binding=0.0.0.0
```

Comment:
this is meant for production too
I don't think rails s actually loads puma.rb ... ran into this issue on
another project ...

On Wed, May 20, 2015 at 8:35 PM, Roman Sandler <notifications@github.com>
wrote:

> In Dockerfile
> <https://github.com/zendesk/samson/pull/381#discussion_r30770667>:
>
> > +ADD plugins /app/plugins
> > +
> > +RUN npm install
> > +
> > +RUN bundle install --without test sqlite postgres --quiet --local --jobs 4 || bundle check
> > +
> > +# Code
> > +ADD config /app/config
> > +ADD app /app/app
> > +ADD lib /app/lib
> > +
> > +RUN DATABASE_URL=mysql2://user:pass@127.0.0.1/null RAILS_ENV=development PRECOMPILE=1 bundle exec rake --trace assets:precompile
> > +
> > +EXPOSE 3000
> > +
> > +CMD DATABASE_URL=$MYSQL_URL bundle exec rails server --binding=0.0.0.0
>
> rails s magically boots puma. We could make it explicit (and use
> config/puma.rb) but I'm not sure if that matters for development
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/381/files#r30770667>.
>